packages/manager/apps/pci-load-balancer/src/api/hook/useAddons/useAddons.ts

```diff
@@ -0,0 +1,80 @@
+import { useMemo } from 'react';
+import {
+  getCatalogQuery,
+  getProductAvailabilityQuery,
+} from '@ovh-ux/manager-pci-common';
+import { useQueries } from '@tanstack/react-query';
+import { sortProductBySize } from '@/utils/utils';
+
+export const useAddons = (
+  ovhSubsidiary: string,
+  projectId: string,
+  addonFamily: string,
+) =>
+  useQueries({
+    queries: [
+      getCatalogQuery(ovhSubsidiary),
+      getProductAvailabilityQuery(projectId, ovhSubsidiary, {
+        addonFamily,
+      }),
+    ],
+    combine: ([
+      { data: catalog, isFetching: isCatalogFetching },
+      { data: plans, isFetching: isPlansFetching },
+    ]) => {
+      let addons = [];
+
+      if (plans?.plans && catalog) {
+        addons = plans.plans.map(({ code, regions }) => {
+          const addon = catalog.addons.find(
+            ({ planCode }) => code === planCode,
+          );
+
+          return {
+            ...addon,
+            regions,
+          };
+        });
+      }
```

Comment:
We could prevent from the mutable array, using `flatMap()`, something like that, but it's up to you :)

```ts
const addons = (plans?.plans ?? []).flatMap(({ code, regions }) => {
  const addon = catalog?.addons.find(({ planCode }) => code === planCode);
  return addon ? [{ ...addon, regions }] : [];
});
```